pkg/kubectl/cmd/apiresources/apiresources.go

```diff
@@ -86,7 +86,7 @@ func NewCmdApiResources(f cmdutil.Factory, ioStreams genericclioptions.IOStreams
 		Long:    "Print the supported API resources on the server",
 		Example: apiresourcesExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmdutil.CheckErr(o.Validate(cmd))
+			cmdutil.CheckErr(o.Validate(cmd, args))
```

Comment:
You need to create `Complete` method and check that there. `Validate` should never accept any parameters, not even `cmd` like here. See other commands for examples.

Reply:
Something we've done to adhere to the pattern @soltysh is describing above, is to save command args to a `[]string` field in the Options struct using the Complete method, then keeping the validation for those args in the `Validate` method by using the `[]string` field in the Options struct